src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs

```diff
@@ -142,129 +143,166 @@ private IEnumerable<DatabaseTable> GetTables(DbConnection connection, Func<Datab
 		}
 	}
 
-	private const string GetColumnsQuery =
-		@"SELECT
-               trim(RF.RDB$FIELD_NAME) as COLUMN_NAME,
-               COALESCE(RF.RDB$DEFAULT_SOURCE, F.RDB$DEFAULT_SOURCE) as COLUMN_DEFAULT,
-               COALESCE(COALESCE(RF.RDB$NULL_FLAG, F.RDB$NULL_FLAG), 0) as NOT_NULL,
-               CASE Coalesce(F.RDB$FIELD_TYPE, 0)
-                WHEN 7 THEN
-                 CASE F.RDB$FIELD_SUB_TYPE
-                  WHEN 0 THEN 'SMALLINT'
-			      WHEN 1 THEN 'NUMERIC(' || (F.RDB$FIELD_PRECISION) || ',' || ABS(F.RDB$FIELD_SCALE) || ')'
-			      WHEN 2 THEN 'DECIMAL(' || (F.RDB$FIELD_PRECISION) || ',' || ABS(F.RDB$FIELD_SCALE) || ')'
-				  ELSE '?'
-                 END
-                WHEN 8 THEN
-                 CASE F.RDB$FIELD_SUB_TYPE
-                  WHEN 0 THEN 'INTEGER'
-			      WHEN 1 THEN 'NUMERIC(' || (F.RDB$FIELD_PRECISION) || ',' || ABS(F.RDB$FIELD_SCALE) || ')'
-			      WHEN 2 THEN 'DECIMAL(' || (F.RDB$FIELD_PRECISION) || ',' || ABS(F.RDB$FIELD_SCALE) || ')'
-				  ELSE '?'
-                 END
+	private void GetColumns(DbConnection connection, IReadOnlyList<DatabaseTable> tables, Func<DatabaseTable, bool> tableFilter)
+	{
+		var getColumnsQuery = $@"
```

Comment:
Put it outside into field/helper method. Maybe even fix other queries with string interpolation, etc. for consistency sake. 

For even more points, use `"""`. ;)

Reply:
Sorry, but I didn’t quite understand that. Could you please clarify? Examples?

Reply:
P.S.: I can refactor ALL of these queries for you, if you allow me😉.

But I will need some guidelines / examples to follow. I can't see a clear pattern as they are, now.

P.S.2.: Regarding `"""`, I fully agree. It's much better.

Reply:
Got it. Something like [this](https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbDataReader.cs#L1133-L1159).

Reply:
Done. Adding final touches.